### DISCUSSION TITLE
Tricks to prevent glitches at the end of every song, caused by liquidsoap upstream?

### DISCUSSION THREAD

**Discussion by unudelaradio** [AzuraCast/AzuraCast#6894] [Q&A]
Hello!

Anyone could help me with some ideas or tricks to avoid the issue with glitches/pauses at the end of every track in AutoDJ? It happens periodically, and the only way i know to solve it is to restart AutoDJ.

As i understood, this is a known issue caused by Liquidsoap upstream. 

Is there any way to avoid or prevent these glitches? Other than manually restarting AutoDJ? I noticed that on the stations with daily live djs, the AutoDJ works fine. I guess that those live streams are like a refresh for AutoDJ, preventing the issue. But when AutoDJ plays without interruptions, for a long time, the problem keep appearing, on all the stations. 

If there's no solution, maybe a custom code for auto-restart the AutoDJ once per week, late at night, will help me. 

Although i hope there's a "silent/delicate" way to solve this, without interrupting the music.

Many thanks!

**Comment by gAlleb** [comment]
I'm not sure glitches/pauses are caused by liquidsoap. They may happen due to CPU peaks or remote music storage (if you are using one). There is a problem with pop/clicks but It seems unrelated here.

Let's see what other folks say)

**Comment by unudelaradio** [comment]
I was referring to this issue: https://github.com/AzuraCast/AzuraCast/issues/6736

I'm surprised that not many users talk about this. To understand better the issue behaviour and how it sounds, take a listen to this recorded file and you will notice the glitches/freezes at the end of every track:

https://drive.google.com/file/d/1zPUMxwFHMSej7zsyC3kYf5TnsO0Gt7hy/

So... i'm looking for a trick to avoid restrarting the AutoDJ on and on and on, every week. Works fine after restart, but then, after few days the glitches are back. I have to monitor continuously every station and if i hear the problem, to manually restart and interrupt the music. :(

There must be another way, an automatic way, to solve this.

**Comment by BusterNeece** [comment]
We've heard from a few users that the crackling audio problem is persisting. If you haven't already, I'd recommend updating to the latest Rolling Release version as it incorporates a newer version of the Liquidsoap rolling release that has some fixes applied to it.

I plan to keep the issue you referenced up until we can conclude more definitively whether this is likely to still be an issue in newer versions of AzuraCast, especially before we tag a new stable release.

**Comment by unudelaradio** [comment]
@BusterNeece I just updated AzuraCast to the latest Rolling Release on one of my servers, few minutes ago. I can confirm that the pops and clicks are still there. So sad! With Stereo Tool enabled, are even more obvious.

But i think there are two different issues:

1. The cracking one, with those pops and clicks, while a track is mixing with another one.
2. The freezing one (which i was referring to) with the pauses, glitches at the end of every track, before the mixing process starts. This issue is not permanent. Appears after a while... maybe a week or two after AutoDJ plays without interruptions. And disappears if the AutoDJ is restarted.

**Reply by BusterNeece** [reply]
I still suspect this is an issue with Liquidsoap's audio processing stack. I've brought the issue back up with them to make sure it's on their radar.

**Comment by ultragain** [comment]
Yes for me its an serious bug so i still using 0.18.4. till the probs are solved.
